pkg/asset/machines/powervs/machines.go

```diff
@@ -0,0 +1,89 @@
+	"github.com/openshift/installer/pkg/types"
+	"github.com/openshift/installer/pkg/types/powervs"
+	machineapi "github.com/openshift/machine-api-operator/pkg/apis/machine/v1beta1"
+	"github.com/pkg/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+// Machines returns a list of machines for a machinepool.
+func Machines(clusterID string, config *types.InstallConfig, pool *types.MachinePool, role, userDataSecret string, userTags map[string]string) ([]machineapi.Machine, error) {
+	if poolPlatform := pool.Platform.Name(); poolPlatform != powervs.Name {
+		return nil, fmt.Errorf("non-PowerVS machine-pool: %q", poolPlatform)
+	}
+	platform := config.Platform.PowerVS
+	mpool := pool.Platform.PowerVS
+
+	total := int64(1)
+	if pool.Replicas != nil {
+		total = *pool.Replicas
+	}
+	var machines []machineapi.Machine
+	for idx := int64(0); idx < total; idx++ {
+		provider, err := provider(clusterID, platform, mpool, userDataSecret, userTags)
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to create provider")
+		}
+		machine := machineapi.Machine{
+			TypeMeta: metav1.TypeMeta{
+				APIVersion: "machine.openshift.io/v1beta1",
+				Kind:       "Machine",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: "openshift-machine-api",
+				Name:      fmt.Sprintf("%s-%s-%d", clusterID, pool.Name, idx),
+				Labels: map[string]string{
+					"machine.openshift.io/cluster-api-cluster":      clusterID,
+					"machine.openshift.io/cluster-api-machine-role": role,
+					"machine.openshift.io/cluster-api-machine-type": role,
+				},
+			},
+			Spec: machineapi.MachineSpec{
+				ProviderSpec: machineapi.ProviderSpec{
+					Value: &runtime.RawExtension{Object: provider},
+				},
+			},
+		}
+		machines = append(machines, machine)
+	}
+	return machines, nil
+}
+
+func provider(clusterID string, platform *powervs.Platform, mpool *powervs.MachinePool, userDataSecret string, userTags map[string]string) (*powervsprovider.PowerVSMachineProviderConfig, error) {
+
+	//Setting only the mandatory parameters
+	config := &powervsprovider.PowerVSMachineProviderConfig{
+		TypeMeta:          metav1.TypeMeta{
+			Kind:       "PowerVSMachineProviderConfig",
+			APIVersion: powervsprovider.GroupVersion.String(),
+		},
+		ObjectMeta:        metav1.ObjectMeta{},
+		Spec:              powervsprovider.PowerVSMachineProviderConfigSpec{},
+		Status:            powervsprovider.PowerVSMachineProviderConfigStatus{},
+		ServiceInstanceID: mpool.ServiceInstance,
+		ImageID:           mpool.ImageID,
+		UserDataSecret:    &corev1.LocalObjectReference{Name: userDataSecret},
+		CredentialsSecret: &corev1.LocalObjectReference{Name: "powervs-credentials-secret"},
+		MachineType:       mpool.SystemType, //TODO(Doubt): should we rename either of the variable for better understanding
+		ProcessorType:     mpool.ProcShare, //TODO(Doubt): should we rename either of the variable for better understanding
```

Comment:
yes, let's use names to match the api descriptions here: https://cloud.ibm.com/apidocs/power-cloud#pcloud-pvminstances-post (so this would be ProcessorType, and the next would be Processors)

Reply:
@mkumatag you decided on the API names right? :D

Reply:
yes, as per our slack discussion we have decided to stick to powervs API definition, @Karthik-K-N can you please send a PR to the cluster-api-provider-powervs with the changes updating to names from https://cloud.ibm.com/apidocs/power-cloud#pcloud-pvminstances-post params?

Reply:
Sure, Will do